pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/filter/BrokerFilter.java

```diff
@@ -42,9 +42,8 @@ public interface BrokerFilter {
      * @param context The load manager context.
      * @return Filtered broker list.
      */
-    Map<String, BrokerLookupData> filter(Map<String, BrokerLookupData> brokers,
-                                         ServiceUnitId serviceUnit,
-                                         LoadManagerContext context)
-            throws BrokerFilterException;
+    CompletableFuture<Map<String, BrokerLookupData>> filter(Map<String, BrokerLookupData> brokers,
+                                                            ServiceUnitId serviceUnit,
+                                                            LoadManagerContext context);
```

Comment:
We should not cherry-pick this public API change to branch-3.0 @Demogorgon314

Reply:
This method has yet to be exposed to users. In this case, can we cherry-pick?

Reply:
@BewareMyPower I pushed a PR to add the broker filter sync method back: https://github.com/apache/pulsar/pull/20826. Is this way acceptable?

Reply:
Yes. It's okay. But since a new method is added into the `BrokerFilter`, we should still avoid cherry-picking it to release branches.